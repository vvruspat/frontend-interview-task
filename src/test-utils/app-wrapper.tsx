import React, { FC } from "react";
import { Provider } from "react-redux";
import { GlobalStyles } from "../components/global-styles";
import { AuthProvider } from "../auth";
import { store } from "../store/store";

export const AppWrapper: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <GlobalStyles />
      <AuthProvider>{children}</AuthProvider>
    </Provider>
  );
};
