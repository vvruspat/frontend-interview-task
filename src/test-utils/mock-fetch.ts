import { error401Data } from "./test-data/error";
import { feedbacksData } from "./test-data/feedbacks";
import { jwtData } from "./test-data/jwt";
import { themesData } from "./test-data/themes";

export function mockFetch() {
  let status: number = 200;

  return {
    setStatus: (newStatus: number) => {
      status = newStatus;
    },

    fetch: jest.fn().mockImplementation((url: string) => {
      if (status !== 200) {
        return Promise.reject(error401Data);
      }

      if (url.indexOf("/api/reviews") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: feedbacksData }),
        });
      }

      if (url.indexOf("/api/themes") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: themesData }),
        });
      }

      if (url.indexOf("/login") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve(jwtData),
        });
      }

      return Promise.resolve({
        ok: true,
        status: status,
        json: () => {},
      });
    }),
  };
}
