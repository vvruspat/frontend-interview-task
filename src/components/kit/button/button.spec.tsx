import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { AppWrapper } from "../../../test-utils/app-wrapper";
import { Button, PrimaryButton, SecondaryButton, OutlineButton } from ".";

describe("Button", () => {
  test("renders button", async () => {
    const onClick = jest.fn();
    const component = render(<Button onClick={onClick}>Test</Button>, {
      wrapper: AppWrapper,
    });
    const button = component.getByText("Test");

    fireEvent(
      button,
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      }),
    );
    expect(button).toHaveTextContent("Test");
    expect(button).toMatchSnapshot();
    expect(onClick).toBeCalled();
  });

  test("renders PrimaryButton", async () => {
    const component = render(<PrimaryButton>Test</PrimaryButton>, {
      wrapper: AppWrapper,
    });
    const button = component.getByText("Test");

    expect(button).toHaveTextContent("Test");
    expect(button).toMatchSnapshot();
  });

  test("renders SecondaryButton", async () => {
    const component = render(<SecondaryButton>Test</SecondaryButton>, {
      wrapper: AppWrapper,
    });
    const button = component.getByText("Test");

    expect(button).toHaveTextContent("Test");
    expect(button).toMatchSnapshot();
  });

  test("renders OutlineButton", async () => {
    const component = render(<OutlineButton>Test</OutlineButton>, {
      wrapper: AppWrapper,
    });
    const button = component.getByText("Test");

    expect(button).toHaveTextContent("Test");
    expect(button).toMatchSnapshot();
  });
});
