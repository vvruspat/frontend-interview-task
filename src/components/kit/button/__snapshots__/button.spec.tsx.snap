// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders OutlineButton 1`] = `
<button
  class="sc-bdfBwQ sc-hKgILt ddILAN glSgeK"
>
  Test
</button>
`;

exports[`Button renders PrimaryButton 1`] = `
<button
  class="sc-bdfBwQ sc-gsTCUz ddILAN fxcCaX"
>
  Test
</button>
`;

exports[`Button renders SecondaryButton 1`] = `
<button
  class="sc-bdfBwQ sc-dlfnbm ddILAN AGrjb"
>
  Test
</button>
`;

exports[`Button renders button 1`] = `
<button
  class="sc-bdfBwQ ddILAN"
>
  Test
</button>
`;
